internal/controllers/gateway/gateway_controller.go

```diff
@@ -286,7 +327,11 @@ func (r *GatewayReconciler) listGatewaysForHTTPRoute(_ context.Context, obj clie
 		return nil
 	}
 	recs := []reconcile.Request{}
-	for _, gateway := range routeAcceptedByGateways(httpRoute.Namespace, httpRoute.Status.Parents) {
+	for _, gateway := range routeAcceptedByGateways(httpRoute) {
+		if !r.GatewayNN.MatchesNN(gateway) {
+			continue
+		}
```

Comment:
This one I'd prefer to keep this way. In case anything needs to be added behind this `if` then nothing needs to be changed w.r.t this bit here.

If we change it as suggested then that `if` also need to change.